### DISCUSSION TITLE
Behavior of session data (`umami.identify`)

### DISCUSSION THREAD

**Discussion by Sleepful** [umami-software/umami#3104] [Q&A]
https://umami.is/docs/tracker-functions#session-data
```js
umami.identify({ email: 'bob@aol.com' });
```

- Does the UI allow to filter sessions by their `identify` properties?
- What happens if I set the same property key twice? Will it erase the first value?
- Is there a way to get the visitor ID from the client?

**Comment by mikecao** [comment]
1. Not yet, we are adding that feature
2. Yes, it will overwrite
3. What client?

**Reply by Sleepful** [reply]
1. Awesome! 
2. Got it. I was looking for a way to append data to the same key on a single session to track visitor requests (http requests on the server) with my own IDs. This idea of appending IDs wouldn't work tho.
3. I was thinking the browser SDK because that would be the place where the visitor is tracked. Although it would be nice to get the visitor ID on the server too.

Ok so basically: 

Each visitor makes requests to the server, on the server I want to track certain actions and I want to correlate them with visitor IDs in Umami.

This would be easy if each visitor was a user with an email, but right now the visitors do not have usernames. Do you suppose I could correlate HTTP requests with visitors IDs? If so, how could I do that?

**Reply by mikecao** [reply]
You could probably use something like this, https://github.com/fingerprintjs/fingerprintjs to generate a unique id for each user.

**Reply by AriDeve** [reply]
I read the comment to confirm that there shouldn't be duplicate properties, and it seems like something is not working correctly with the overwriting of properties.
I have just created an issue #3121
